### PR TITLE
Add message to SolidusShipStation::Api::RequestError

### PR DESCRIPTION
**Pull request by nbelzer**
The commit message explains it best:

Sentry uses `.message`[^1] to set the value on their error messages. The method exists by default on `Exception` (from which `RequestError` inherits).  By default it returns the message provided to the initialiser of `RequestError`.  This PR introduces better formatting of this message to allow it to be picked up better by Sentry.

[^1]: https://github.com/getsentry/sentry-ruby/blob/94c28fe9a7fd1c15a10eeeb0e1c4fcac099d4166/sentry-ruby/lib/sentry/interfaces/single_exception.rb#L7

### CODE REVIEW

**Review by waiting-for-dev — APPROVED** [review]
Thanks! That's really useful. Do you think using the same message for the Ruby exception itself would make sense? Not it's using the unparsed `@response_body`.

**Comment by nbelzer** [comment]
@waiting-for-dev I agree it would be better, it should be more in line with other types of error messages. But it could be a breaking change for some people, if they rely on this message somehow.  I'd be happy to change it and add a small notice to the changelog if you are okay with this.

Just to confirm you mean changing:
https://github.com/solidusio-contrib/solidus_shipstation/blob/f70e1dc44319464e5c15f608cd0fa5ec2a84eb37/lib/solidus_shipstation/api/request_error.rb#L29

to:
```ruby
super(message)
```

**Comment by waiting-for-dev** [comment]
> @waiting-for-dev I agree it would be better, it should be more in line with other types of error messages. But it could be a breaking change for some people, if they rely on this message somehow. I'd be happy to change it and add a small notice to the changelog if you are okay with this.
> 
> Just to confirm you mean changing:
> 
> https://github.com/solidusio-contrib/solidus_shipstation/blob/f70e1dc44319464e5c15f608cd0fa5ec2a84eb37/lib/solidus_shipstation/api/request_error.rb#L29
> 
> to:
> 
> ```ruby
> super(message)
> ```

Yeah, I mean precisely that. And you're right, that would be a breaking change. If you wish, we could prepare the change in another PR and label it for v2.0. What do you think? Pinging @kennyadsl to be sure we aren't missing anything, but, as far as I know, extensions should also follow semver.

**Comment by nbelzer** [comment]
@waiting-for-dev That sounds like a good plan, I'll prepare another PR with that change exactly and a note in the changelog on how to handle the breaking change. Should be as simple as to use `error.response_body` in scenarios where the message is used instead.

**Comment by nbelzer** [comment]
After further inspection it actually looks like this change will already introduce the breaking change.  By default `.message` (on the `Exception` class) returns the name of the exception or the message that was passed to the initialiser.

My proposed change here overrides `.message` so it no longer matters what we pass to the `super` initialiser.  Since this is intended we should mark this PR as breaking change.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,19 @@
 # Changelog
 
+## v2.0.0
+
+**Breaking changes**
+
+- The default message for a `RequestError` has been changed from the response body of the response instead include a parsed version of this body.  If your code depends on the response body you need to access it through the `.response_body` attribute on the error instead.
```
Comment (waiting-for-dev) [inline CHANGELOG.md:7]: I think that the wording is a bit confusing. Something like the following could be clearer:

```suggestion
The default message for a `RequestError` has been changed from the raw response body to a parsed version. If your code depends on the response body, you need to access it through the `.response_body` attribute on the exception instead.
```

File: lib/solidus_shipstation/api/request_error.rb [hunk 2]
```diff
@@ -26,7 +26,18 @@ def initialize(response_code:, response_body:, response_headers:)
...
         @response_headers = response_headers
 
-        super(response_body)
+        super(message_from_body)
+      end
+
+      private
+
+      def message_from_body
+        return nil unless @response_body
```
Comment (waiting-for-dev) [inline lib/solidus_shipstation/api/request_error.rb:35]:
```suggestion
        return unless @response_body
```

**Review by waiting-for-dev — APPROVED** [review]
Two minor suggestions.

**Comment by waiting-for-dev** [comment]
> My proposed change here overrides .message so it no longer matters what we pass to the super initializer. Since this is intended we should mark this PR as breaking change.

If I get it well, as this is an error class that users shouldn't raise themselves, the BC goes more along with what you wrote in the README changes. If it were another type of error class, I wouldn't worry a lot about it. However, it's plausible that people are parsing the response body themselves.

**Comment by nbelzer** [comment]
I agree with your suggestions and have included them in the commits.

> If I get it well, as this is an error class that users shouldn't raise themselves, the BC goes more along with what you wrote in the README changes. If it were another type of error class, I wouldn't worry a lot about it. However, it's plausible that people are parsing the response body themselves.

Exactly, I think it is possible some people might be doing just that for their configuration of the `error_handler`.

**Comment by waiting-for-dev** [comment]
Thanks, @nbelzer! We've agreed with @kennyadsl to release v2.0 after we merge it.

**Pull request merged**